modules/platforms/cpp/ignite/common/config.h

```diff
@@ -51,10 +51,12 @@
 #endif
 
 /**
- * Macro IGNITE_SWITCH_WIN_OTHER that uses first option on Windows and second on any other OS.
+ * Macro IGNITE_OS_SWITCH that uses first option on Windows, second on Linux and third on MacOS.
```

Comment:
As we add support for more OSes this might get rather unwieldy. I think we should get rid of this and find something more friendly for future extensions.

Reply:
Probably, yes, but I don't know what other operating systems are we going to support, this looks like an exhaustive list (at least for now). z/OS probably? Frankly, I don't have enough experience with C++ to come up with a better idea than this macro

Reply:
I'm not sure there going to be a lot of other platforms, actually. I would not remove `IGNITE_SWITCH_WIN_OTHER` macro while adding `IGNITE_OS_SWITCH` by the way, as there are a lot of places where there is the same code for Linux and macOS, so it's still very usable in most places. Anyway, even `IGNITE_OS_SWITCH` is not a big deal currently, and for the future - we will see.